src/main/kotlin/com/soywiz/korge/intellij/editor/tiled/TiledMap.kt

```diff
@@ -0,0 +1,378 @@
+
+import com.soywiz.klogger.*
+import com.soywiz.korge.view.tiles.*
+import com.soywiz.korim.bitmap.*
+import com.soywiz.korim.color.*
+import com.soywiz.korma.geom.*
+import java.util.*
+import kotlin.collections.LinkedHashMap
+
+class TiledMapData(
+	var orientation: TiledMap.Orientation = TiledMap.Orientation.ORTHOGONAL,
+	//TODO: support render order
+	var renderOrder: TiledMap.RenderOrder = TiledMap.RenderOrder.RIGHT_DOWN,
+	var compressionLevel: Int = -1,
+	var width: Int = 0,
+	var height: Int = 0,
+	var tilewidth: Int = 0,
+	var tileheight: Int = 0,
+	var hexSideLength: Int? = null,
+	var staggerAxis: TiledMap.StaggerAxis? = null,
+	var staggerIndex: TiledMap.StaggerIndex? = null,
+	var backgroundColor: RGBA? = null,
+	var nextLayerId: Int = 1,
+	var nextObjectId: Int = 1,
+	var infinite: Boolean = false,
+	val properties: MutableMap<String, TiledMap.Property> = mutableMapOf(),
+	val allLayers: MutableList<TiledMap.Layer> = arrayListOf(),
+	val tilesets: MutableList<TileSetData> = arrayListOf(),
+	var editorSettings: EditorSettings? = null
+) {
+	val pixelWidth: Int get() = width * tilewidth
+	val pixelHeight: Int get() = height * tileheight
+	inline val tileLayers get() = allLayers.tiles
+	inline val imageLayers get() = allLayers.images
+	inline val objectLayers get() = allLayers.objects
+
+	val maxGid get() = tilesets.map { it.firstgid + it.tileCount }.max() ?: 0
+
+	fun getObjectByName(name: String) = objectLayers.mapNotNull { it.getByName(name) }.firstOrNull()
+
+	fun clone() = TiledMapData(
+		orientation, renderOrder, compressionLevel,
+		width, height, tilewidth, tileheight,
+		hexSideLength, staggerAxis, staggerIndex,
+		backgroundColor, nextLayerId, nextObjectId, infinite,
+		LinkedHashMap(properties),
+		allLayers.map { it.clone() }.toMutableList(),
+		tilesets.map { it.clone() }.toMutableList()
+	)
+}
+
+fun TiledMap.Object.getPos(map: TiledMapData) = Point(bounds.x / map.tilewidth, bounds.y / map.tileheight)
+
+fun TiledMapData.getObjectPosByName(name: String) = getObjectByName(name)?.getPos(this)
+
+data class TerrainData(
+	val name: String,
+	val tile: Int,
+	val properties: Map<String, TiledMap.Property> = mapOf()
+)
+
+data class AnimationFrameData(
+	val tileId: Int, val duration: Int
+)
+
+data class TerrainInfo(val info: List<Int?>) {
+	operator fun get(x: Int, y: Int): Int? = if (x in 0..1 && y in 0..1) info[y * 2 + x] else null
+}
+
+class WangSet(
+	val name: String,
+	val tileId: Int,
+	val properties: Map<String, TiledMap.Property> = mapOf(),
+	val cornerColors: List<WangColor> = listOf(),
+	val edgeColors: List<WangColor> = listOf(),
+	val wangtiles: List<WangTile> = listOf()
+) {
+	class WangColor(
+		val name: String,
+		val color: RGBA,
+		val tileId: Int,
+		val probability: Double = 0.0
+	)
+
+	class WangTile(
+		val tileId: Int,
+		val wangId: Int,
+		val hflip: Boolean = false,
+		val vflip: Boolean = false,
+		val dflip: Boolean = false
+	)
+}
+
+class EditorSettings(
```

Comment:
Same here. Can you rename it to `TiledEditorSettings`?

Reply:
I think I will probably move this class inside TiledMap